ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatRecord.java

```diff
@@ -362,6 +399,192 @@ private void verifyCrc() throws MessageFormatException {
   }
 
   /**
+   *
+   *  - - - - - - - - - - - - - - - - - - -- - -- - - - - - - - - - - - - -  - - - - - - - - - - - - - - - - - - - -
+   * |         |              |                 |               |           |               |           |           |
+   * | version | payload size | Blob Encryption | Blob Property | Delete    | User Metadata | Blob      | Crc       |
+   * |(2 bytes)|   (8 bytes)  | Key Relative    | Relative      | Relative  | Relative      | Relative  | (8 bytes) |
+   * |         |              | Offset          | Offset        | Offset    | Offset        | Offset    |           |
+   * |         |              | (4 bytes)       | (4 bytes)     | (4 bytes) | (4 bytes)     | (4 bytes) |           |
+   * |         |              |                 |               |           |               |           |           |
+   *  - - - - - - - - - - - - - - - - - - -- - -- - - - - - - - - - - - - -  - - - - - - - - - - - - - - - - - - - -
+   *
+   *  version         - The version of the message header
+   *
+   *  payload size    - The size of the message payload.
+   *                    Blob Encryption Key Record Size (if present) + (Blob prop record size or delete record size) +
+   *                    user metadata size + blob size
+   *
+   *  Blob Encryption - The offset at which the blob encryption key record is located relative to this message.
+   *  Key relative      Non-existence of blob key record is indicated by -1. Blob Keys are optionally present for Put
+   *  offset            records. Blob Keys will be absent for Delete records.
+   *
+   *  blob property   - The offset at which the blob property record is located relative to this message. Only one of
+   *  relative offset   blob property/delete relative offset field can exist. Non existence is indicated by -1
+   *
+   *  delete          - The offset at which the delete record is located relative to this message. Only one of blob
+   *  relative offset   property/delete relative offset field can exist. Non existence is indicated by -1
+   *
+   *  user metadata   - The offset at which the user metadata record is located relative to this message. This exist
+   *  relative offset   only when blob property record and blob record exist
+   *
+   *  blob metadata   - The offset at which the blob record is located relative to this message. This exist only when
+   *  relative offset   blob property record and user metadata record exist
+   *
+   *  crc             - The crc of the message header
+   *
+   */
+  public static class MessageHeader_Format_V2 {
+    private ByteBuffer buffer;
+
+    // total size field start offset and size
+    public static final int Total_Size_Field_Offset_In_Bytes = Version_Field_Size_In_Bytes;
```

Comment:
I guess we follow the convention of ALL_CAPS for static final fields.

Reply:
I don't feel that this is a change we should make at this time. We should probably create an issue to do this across the codebase. I have followed the existing convention in this file.